composeApp/src/commonMain/kotlin/org/ooni/probe/domain/GetTestDescriptorsBySpec.kt

```diff
@@ -0,0 +1,51 @@
+package org.ooni.probe.domain
+
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.first
+import org.ooni.probe.data.models.Descriptor
+import org.ooni.probe.data.models.RunSpecification
+
+// Filter TestDescriptors and netTests based on the specification provided
+class GetTestDescriptorsBySpec(
+    private val getTestDescriptors: () -> Flow<List<Descriptor>>,
+) {
+    suspend operator fun invoke(spec: RunSpecification): List<Descriptor> =
+        getTestDescriptors()
+            .first()
+            .filterNot { it.isExpired }
```

Comment:
We dont want to depend on this property as it is not assured that it returns the correct status always.

> If the descriptor is fetched in an update and the expiry date is set to tomorrow, `is_expired` will be false. the following day the descriptor will have expired but since a new version is not available in the backend, expired status remains in the cached copy as `false`

Reply:
Ah, got it. But if do fetch `isExpire=true` it's definitely expired? We don't have an `expiresAt` timestamp do we?

Reply:
Yeah. If expired is set to true, its for sure expired.